### DISCUSSION TITLE
How to work with cookies

### DISCUSSION THREAD

**Discussion by skreutzberger** [usebruno/bruno#410] [Q&A] [answered]
Hi, we are using cookies for authorization. So the normal flow is that a login request gets a cookie as response and all API requests should then automatically use the cookie by sending them in the header on each request. That is a key feature of Insomnia.

I tried the same flow with Bruno but it seems that cookies are not stored between requests. Is there a workaround? Like for example storing a response header value in a variable and then use the variable's value at another request's header automatically?

**Comment by helloanoop** [comment, marked as answer]
Hi @skreutzberger 

> Is there a workaround? Like for example storing a response header value in a variable and then use the variable's value at another request's header automatically?

You can refer this script https://github.com/usebruno/bruno/discussions/385#discussioncomment-7206789

This can be a tad bit frustrating to have this pre-request script in your every request.
I am currently working on getting a collection level script support that can do this for all requests https://github.com/usebruno/bruno/issues/334

**Reply by alberthammerich** [reply]
Any updates on this? 🙌

**Reply by FallenHero66** [reply]
> Any updates on this? 🙌

since I was also just digging around in this area: commit e1a96e0 added support for cookies (the Set-Cookie header is honored and cookies are saved and sent along with POST requests automatically). If you don't have support for cookies yet, you might be running in the same issue i was: the apt package of bruno is behind ~6 months on releases (v1.12.3 instead of 1.28.0), updating to the most recent version brought cookie support.
Downside: deleting cookies is not added yet, but you can deactivate them in the preferences altogether.

**Comment by danielo515** [comment]
I'm on 1.39.0, macos and it seems cookies are not automatically sent for get requests. Is there a way I can validate this?

**Reply by aidenberzins** [reply]
Having a similar issue. Trying to determine if it is a me issue or Bruno

**Reply by helloanoop** [reply]
@aidenberzins Can you confirm the set-cookie requests coming in response. You can inspect it in devtools.
<img width="702" height="551" alt="image" src="https://github.com/user-attachments/assets/fa37d791-93c5-4c61-b164-6f6e339c77d8" />

Also, you can explore the cookie jar (clicking on cookies button in bottom right corner)
<img width="1387" height="748" alt="image" src="https://github.com/user-attachments/assets/681e25ae-34f1-40d1-bb60-fd5f724891aa" />

**Reply by aidenberzins** [reply]
I am able to confirm that bruno is capturing the cookie but it doesn't seem to be attaching it to the req headers when sending requests.  

<img width="2324" height="830" alt="Screenshot 2025-10-27 at 11 50 56 AM" src="https://github.com/user-attachments/assets/9471364f-0531-4ccf-8751-dc6c647505d1" />
<img width="2135" height="352" alt="Screenshot 2025-10-27 at 11 49 35 AM" src="https://github.com/user-attachments/assets/01955ac3-1f23-4d82-9477-71a57873b0d4" />

**Reply by helloanoop** [reply]
@skreutzberger The screenshot you shared is for the response headers. Could you look at the request headers in the Timeline Tab ?

**Reply by aidenberzins** [reply]
Looks like GET request sends the cookie. 
<img width="514" height="278" alt="Screenshot 2025-10-27 at 3 00 36 PM" src="https://github.com/user-attachments/assets/60699267-f235-4da7-b49c-c714a0ec31ce" />

Apologies not sure if it was user error before or if there was some bug. I will keep tabs on it and steps to reproduce if it happens again.

**Reply by helloanoop** [reply]
Cool !

**Comment by mgrandi** [comment]
cookies are also not being added by requests, well rather they stored per request but what is the use of having the cookies saved if i can't use them in a subsequent request?

**Comment by MathieuDR** [comment]
Would be nice if I could remove the cookie / header in my collection pre-request script.

**Comment by danielatsellhub** [comment]
Hello everyone, any way to get this working?

**Comment by khaneh-liner** [comment]
I found that a cookie is not sent when there is a leading dot in the domain.
